src/common/utxobased/db/makeBaselets.ts

```diff
@@ -102,37 +117,37 @@ export interface Databases {
 }
 
 export interface AddressTables {
-  addressByScriptPubkey: HashBase
+  addressByScriptPubkey: HashBase<IAddress>
+  scriptPubkeyByPath: CountBase<string>
+  lastUsedByFormatPath: HashBase<number>
   // deprecated
   addressPathByMRU: CountBase
-  scriptPubkeyByPath: CountBase
   // deprecated
   scriptPubkeysByBalance: RangeBase
   // deprecated
   usedFlagByScriptPubkey: HashBase
-  lastUsedByFormatPath: HashBase
 }
 
 export interface TransactionTables {
-  txById: HashBase
+  txById: HashBase<IProcessorTransaction>
+  txIdsByBlockHeight: RangeBase<TxIdsByBlockHeight>
+  txIdsByDate: RangeBase<TxIdByDate, 'date', 'txId'>
   // deprecated
   txsByScriptPubkey: HashBase
-  txIdsByBlockHeight: RangeBase
   // deprecated - use txIdsByDate instead
-  txsByDate: RangeBase
-  txIdsByDate: RangeBase
+  txsByDate: RangeBase<IProcessorTransaction>
 }
 
 export interface UTXOTables {
-  utxoById: HashBase
+  utxoById: HashBase<IUTXO>
   // deprecated
-  utxoIdsByScriptPubkey: HashBase
+  utxoIdsByScriptPubkey: HashBase<string>
```

Comment:
Type should be `HashBase<string[]>`. This will incur a necessary fix in the implemenatation.